tests/integration/conftest.py

```diff
@@ -70,15 +71,17 @@ class TestLMDBEnv:
     def __init__(self, *args, **kwargs):
         self.db_path = args[0]
         self.open = partial(self.LMDB_OPEN_FUNC, *args, **kwargs)
+        self.thread_lock = threading.Lock()
 
     @contextmanager
     def begin(self, *args, **kwargs):
         try:
+            self.thread_lock.acquire()
```

Comment:
Should the acquisition of the lock be outside of the try-finally block? i.e. we don't want to call `release()` if there was a problem acquiring the lock...

```suggestion
        self.thread_lock.acquire()
        try:
```

Reply:
https://docs.python.org/3/library/threading.html#using-locks-conditions-and-semaphores-in-the-with-statement describe exactly what Derek propose. But perhaps it'd be even better to use the lock as a context manager, as the docs show.

Reply:
ah, even better - 👍